### ISSUE TITLE
BUG: Config.NUM_EXAMPLES does not exist.

### ISSUE THREAD

**Issue by danglotb** [tech-srl/code2vec#11]
Hello,

I tried to play with Code2Vec, and specialized an already trained model with a my own dataset.

As suggested in #5, I used the following command line:

```sh
python3 -u code2vec.py --load ${trained_model} --data ${data} --test ${test_data} --save ${model_dir}/saved_model
```

However, I have the following errors:

```txt
Traceback (most recent call last):
  File "code2vec.py", line 38, in <module>
    model.train()
  File "/tmp/code2vec/model.py", line 72, in train
    self.config.NUM_EXAMPLES / self.config.BATCH_SIZE * self.config.SAVE_EVERY_EPOCHS), 1)
AttributeError: 'Config' object has no attribute 'NUM_EXAMPLES'
```
The field `NUM_EXAMPLES` does not exist in Config object. I hacker `model.py` and replace this value by `1` (I don't know what I'm doing, but just to see what happen! :-))

It seems that this field is also used in `PathContextReader`
```txt
Starting training
Traceback (most recent call last):
  File "code2vec.py", line 38, in <module>
    model.train()
  File "/tmp/code2vec/model.py", line 76, in train
    config=self.config)
  File "/tmp/code2vec/PathContextReader.py", line 15, in __init__
    self.batch_size = config.TEST_BATCH_SIZE if is_evaluating else min(config.BATCH_SIZE, config.NUM_EXAMPLES)
AttributeError: 'Config' object has no attribute 'NUM_EXAMPLES'
```

According to the code in [`common.py`](https://github.com/tech-srl/code2vec/blob/master/common.py#L33), there is no such field in the `Config` object:

```python
 def __init__(self):
        self.NUM_EPOCHS = 0
        self.SAVE_EVERY_EPOCHS = 0
        self.BATCH_SIZE = 0
        self.TEST_BATCH_SIZE = 0
        self.READING_BATCH_SIZE = 0
        self.NUM_BATCHING_THREADS = 0
        self.BATCH_QUEUE_SIZE = 0
        self.TRAIN_PATH = ''
        self.TEST_PATH = ''
        self.MAX_CONTEXTS = 0
        self.WORDS_VOCAB_SIZE = 0
        self.TARGET_VOCAB_SIZE = 0
        self.PATHS_VOCAB_SIZE = 0
        self.EMBEDDINGS_SIZE = 0
        self.SAVE_PATH = ''
        self.LOAD_PATH = ''
        self.MAX_TO_KEEP = 0
        self.RELEASE = False
        self.EXPORT_CODE_VECTORS = False
```

What would be the default value of this field?

**Comment by urialon** [comment]
Hi Benjamin,
Yes, you're right. This value is loaded from the original training set, but in the case of re-training it is not loaded. I need to think about how to fix this without breaking the previously saved models.
This parameter is simply used to let the training loop know when a full epoch has reached and it should pause training and evaluate on the validation set

In the meantime, it will be easy to fix if you just count the lines in your new training set (using `wc -l my_new_dataset.train.c2v`) and set this value in `Config.py`: add the line `self.NUM_EXAMPLES = 10000` to the `__init__` method of Config (where 10000 is the result of the `wc -l` command).

Please let me know how it goes.

**Comment by danglotb** [comment]
Okay, I'll try this ASAP and keep you update.

Thank you.

**Comment by urialon** [comment]
Hi Benjamin,
I just submitted a fix (16e30715574f44f08164a57e149c9a0accb3679c) which automatically initializes NUM_EXAMPLES as the number of examples in the new dataset.

**Comment by urialon** [comment]
Thanks for pointing out this bug!
